frameworks/Java/armeria/src/main/java/hello/services/PostgresDbService.java

```diff
@@ -0,0 +1,209 @@
+import com.linecorp.armeria.server.annotation.Default;
+import com.linecorp.armeria.server.annotation.Get;
+import com.linecorp.armeria.server.annotation.Param;
+import com.linecorp.armeria.server.annotation.ProducesJson;
+
+public class PostgresDbService
+{
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  private static final DateFormat DATE_FORMATTER = new SimpleDateFormat(
+      "E, dd MMM yyyy HH:mm:ss z");
+
+  private static final String DATABASE_HOST     = "jdbc:postgresql://tfb-database:5432/hello_world";
+  private static final String DATABASE_USER     = "benchmarkdbuser";
+  private static final String DATABASE_PASSWORD = "benchmarkdbpass";
+
+  private static final int    NUMBER_OF_RECORDS = 10000;
+  private static final String SELECT_QUERY      = "SELECT * FROM world WHERE id = ?";
+  private static final String UPDATE_QUERY      = "UPDATE world SET randomNumber = ? WHERE id = ?";
+
+  @Get("/db") @ProducesJson public HttpResponse db()
+  {
+    try
+    {
+      HttpHeaders headers = HttpHeaders.of(HttpStatus.OK).add(
+          HttpHeaderNames.SERVER, "armeria").add(HttpHeaderNames.DATE,
+          DATE_FORMATTER.format(new Date())).contentType(
+          MediaType.JSON_UTF_8);
+
+      return HttpResponse.of(headers, HttpData.of(MAPPER.writeValueAsBytes(
+          getWorld(getRandomNumber(1, NUMBER_OF_RECORDS)))));
+    }
+    catch (Exception e)
+    {
+      return HttpResponse.of(HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  // need to use regex as /queries/{count} doesnt work when count is null
+  @Get("regex:^/queries/(?<count>.*)$") @ProducesJson public HttpResponse queries(
+      @Param("count") @Default("") String count)
+  {
+    try
+    {
+      HttpHeaders headers = HttpHeaders.of(HttpStatus.OK).add(
+          HttpHeaderNames.SERVER, "armeria").add(HttpHeaderNames.DATE,
+          DATE_FORMATTER.format(new Date())).contentType(
+          MediaType.JSON_UTF_8);
+
+      return HttpResponse.of(headers, HttpData.of(
+          MAPPER.writeValueAsBytes(getWorlds(getSanitizedCount(count)))));
+    }
+    catch (JsonProcessingException e)
+    {
+      return HttpResponse.of(HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  @Get("regex:^/updates/(?<count>.*)$") @ProducesJson public HttpResponse update(
+      @Param("count") @Default("") String count)
+  {
+    try
+    {
+      HttpHeaders headers = HttpHeaders.of(HttpStatus.OK).add(
+          HttpHeaderNames.SERVER, "armeria").add(HttpHeaderNames.DATE,
+          DATE_FORMATTER.format(new Date())).contentType(
+          MediaType.JSON_UTF_8);
+
+      return HttpResponse.of(headers, HttpData.of(MAPPER.writeValueAsBytes(
+          getUpdatedWorlds(getSanitizedCount(count)))));
+
+    }
+    catch (JsonProcessingException e)
+    {
+      return HttpResponse.of(HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  private int getRandomNumber(int min, int max)
+  {
+    Random r = new Random();
```

Comment:
You generally want to reuse instances of `Random` rather than creating a new instance for each number that you generate.  And if the `Random` is going to be used by many threads, which I think this is, then that's what `ThreadLocalRandom` is designed for.  That class also conveniently has a `nextInt(minInclusive, maxExclusive)` function, unlike `Random`.  AFAIK most of the other Java frameworks in TFB are using `ThreadLocalRandom` for this.

Here's one example:
https://github.com/TechEmpower/FrameworkBenchmarks/blob/cffabe9d29f4bf4b3a9ad845d70726a8ac87fd11/frameworks/Java/undertow/src/main/java/hello/HelloWebServer.java#L254-L256